### PR TITLE
Autosuggest Ticker Symbols for Network form (user initiated)

### PR DESCRIPTION
**Pull request by vthomas13**
## **Description**
We want to show a "Suggested ticker symbol" as user enters in their ChainID in the Network form, which should help provide additional context to keep users from getting scammed. 

## **Related issues**

Fixes: [#1488](https://github.com/MetaMask/MetaMask-planning/issues/1488)

## **Manual testing steps**
1. Open extension
2. Go to the Settings > Networks
3. Add New Network or Edit an existing network
4. Type 137 into chain id field make sure that the "Suggested ticker symbol" changes dynamically as you type.
5. Type 'ETH' in the Currency Symbol field, and the mismatch warning and suggested ticker should show as 'MATIC'.
6. Click the Suggested ticker symbol once and the "Currency symbol" textfield in the form should auto-populate as 'MATIC'.
7. Continue to edit the Currency Symbol field, the mismatch warning should show again.

### **After**

https://github.com/MetaMask/metamask-extension/assets/10986371/184277c4-4965-497c-bf32-8675cd101ab2

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/settings/networks-tab/networks-form/networks-form.js [hunk 1]
```diff
@@ -213,6 +243,27 @@ const NetworksForm = ({
     dispatch,
   ]);
 
+  const autoSuggestTicker = useCallback((formChainId) => {
+    const decimalChainId = getDisplayChainId(formChainId);
+    if (
+      !decimalChainId ||
```
Comment (darkwing) [inline ui/pages/settings/networks-tab/networks-form/networks-form.js:249]: What condition does this cover?  `0`?

File: ui/pages/settings/networks-tab/networks-form/networks-form.js [hunk 2]
```diff
@@ -682,28 +721,64 @@ const NetworksForm = ({
...
+          helpText={
+            suggestedTicker ? (
+              <Text
+                as="span"
+                variant={TextVariant.bodySm}
+                color={TextColor.textDefault}
+              >
+                {t('suggestedTokenSymbol')}{' '}
+                <Text
+                  as="span"
```
Comment (darkwing) [inline ui/pages/settings/networks-tab/networks-form/networks-form.js:742]: Let's make this a `button` to ensure keyboard accessibility.  Let's also ensure it has the `pointer` cursor.

**Review by darkwing — CHANGES_REQUESTED** [review]
Nice, just a nit or two!

File: ui/pages/settings/networks-tab/networks-form/networks-form.js (hunk 1 above)
Comment (vthomas13) [reply ui/pages/settings/networks-tab/networks-form/networks-form.js]: Good catch, I think getDisplayChainId already accounts for this condition

File: ui/pages/settings/networks-tab/networks-form/networks-form.js [hunk 3]
```diff
@@ -682,28 +729,65 @@ const NetworksForm = ({
...
+        <FormTextField
+          data-testid="network-form-ticker"
+          helpText={
+            suggestedTicker ? (
+              <Text
+                as="span"
+                variant={TextVariant.bodySm}
+                color={TextColor.textDefault}
+              >
+                {t('suggestedTokenSymbol')}{' '}
```
Comment (NidhiKJha) [inline ui/pages/settings/networks-tab/networks-form/networks-form.js:748]: Nit:
```suggestion
                {t('suggestedTokenSymbol')}
```

File: ui/pages/settings/networks-tab/networks-form/networks-form.js [hunk 4]
```diff
@@ -682,28 +729,65 @@ const NetworksForm = ({
...
           value={ticker}
           disabled={viewOnly}
         />
+        {warnings.ticker?.msg ? (
+          <HelpText
+            severity={HelpTextSeverity.Warning}
+            marginTop={1}
+            data-testid="network-form-ticker-warning"
+          >
+            {warnings.ticker?.msg}
```
Comment (NidhiKJha) [inline ui/pages/settings/networks-tab/networks-form/networks-form.js:788]: At this point, we shouldn't check it
```suggestion
            {warnings.ticker.msg}
```

**Review by NidhiKJha — COMMENTED** [review]
Looking really good. Just left some non blocking comments

**Review by NidhiKJha — APPROVED** [review]
LGTM

**Review by darkwing — APPROVED** [review]

**Pull request merged**